File: NXNavigationExtensionDemo/UIKit/Feature/Basic/Controllers/ViewController01_BackgroundColor.m
//
//  ViewController01_BackgroundColor.m
//  NXNavigationExtensionDemo
//
//  Created by Leo Lee on 2020/10/25.
//

#import <NXNavigationExtension/NXNavigationExtension.h>

#import "ViewController01_BackgroundColor.h"
#import "UIColor+RandomColor.h"

@interface ViewController01_BackgroundColor ()

@end

@implementation ViewController01_BackgroundColor

- (UIStatusBarStyle)preferredStatusBarStyle {
    return UIStatusBarStyleLightContent;
}

- (UIColor *)nx_barTintColor {
    return [UIColor whiteColor];
}

- (NSDictionary<NSAttributedStringKey,id> *)nx_titleTextAttributes {
    return @{NSForegroundColorAttributeName: [self nx_barTintColor]};
}

- (UIColor *)nx_navigationBarBackgroundColor {
    return self.randomColor;
}

@end

File: Examples/NXNavigationExtensionDemo/UIKit/Basic/ViewController01_BackgroundColor.swift
//
//  ViewController01_BackgroundColor.swift
//  NXNavigationExtensionDemo
//
//  Created by lidan on 2021/11/8.
//

import NXNavigationExtension
import UIKit

class ViewController01_BackgroundColor: CustomTableViewController {
    override var preferredStatusBarStyle: UIStatusBarStyle {
        return .lightContent
    }
}

extension ViewController01_BackgroundColor {
    override var nx_barTintColor: UIColor? {
        return .white
    }

    override var nx_titleTextAttributes: [NSAttributedString.Key: Any]? {
        return [NSAttributedString.Key.foregroundColor: nx_barTintColor ?? .white]
    }

    override var nx_navigationBarBackgroundColor: UIColor? {
        return randomColor
    }
}
